src/black/__init__.py

```diff
@@ -363,6 +363,17 @@ def target_version_option_callback(
     return [TargetVersion[val.upper()] for val in v]
 
 
+def validate_regex(
+    ctx: click.Context,
+    param: Union[click.Option, click.Parameter],
```

Comment:
`click.Option` is a subclass of `click.Parameter` so I don't think `click.Option` is necessary.
https://github.com/pallets/click/blob/master/src/click/core.py#L2133

```suggestion
    param: click.Parameter,
```